client/src/main/java/org/apache/nemo/client/JobLauncher.java

```diff
@@ -281,7 +284,9 @@ public static void launchDAG(final IRDAG dag,
 
     LOG.info("Launching DAG...");
     serializedDAG = Base64.getEncoder().encodeToString(SerializationUtils.serialize(dag));
-    jobDoneLatch = new CountDownLatch(1);
+    if (jobDoneLatch == null || jobDoneLatch.getCount() == 0) {
```

Comment:
Could you please explain when `jobDoneLatch.getCount()` becomes 0 here? and when `jobDoneLatch` could be null?

Reply:
I'll add comments to explain this bit. Thanks!